runtime/src/asset.rs

```diff
@@ -1962,4 +2035,96 @@ impl<T: Trait> Module<T> {
 
         Ok(())
     }
+
+    pub fn verify_restriction(
+        ticker: &Ticker,
+        extension_caller: T::AccountId,
+        from_did: Option<IdentityId>,
+        to_did: Option<IdentityId>,
+        value: T::Balance,
+        dest: T::AccountId,
+    ) -> RestrictionResult {
+        // 4 byte selector of verify_transfer - 0xD9386E41
+        let selector = hex!("D9386E41");
+        let balance_to = match to_did {
+            Some(did) => T::Balance::encode(&<BalanceOf<T>>::get((ticker, &did))),
+            None => T::Balance::encode(&(0.into())),
+        };
+        let balance_from = match from_did {
+            Some(did) => T::Balance::encode(&<BalanceOf<T>>::get((ticker, &did))),
+            None => T::Balance::encode(&(0.into())),
+        };
+        let encoded_to = Option::<IdentityId>::encode(&to_did);
+        let encoded_from = Option::<IdentityId>::encode(&from_did);
+        let encoded_value = T::Balance::encode(&value);
+        let total_supply = T::Balance::encode(&<Tokens<T>>::get(&ticker).total_supply);
+
+        // Creation of the encoded data for the verifyTransfer function of the extension
+        // i.e fn verify_transfer(
+        //        from: IdentityId,
+        //        to: IdentityId,
+        //        value: Balance,
+        //        balance_from: Balance,
+        //        balance_to: Balance,
+        //        total_supply: Balance
+        //    ) -> RestrictionResult { }
+
+        let encoded_data = [
+            &selector[..],
+            // &encoded_from[..],
+            // &encoded_to[..],
+            &encoded_value[..],
+            &balance_from[..],
+            &balance_to[..],
+            &total_supply[..],
+        ]
+        .concat();
+
+        // Calling extension to verify the compliance rule
+        // native currency value should be `0` as no funds need to transfer to the smart extension
+        // We are passing arbitrary high `gas_limit` value to make sure extension's function execute successfully
+        // TODO: Once gas estimate function will be introduced, arbitrary gas value will be replaced by the estimated gas
+        let is_allowed =
+            Self::call_extension(extension_caller, dest, 0.into(), 5000000, encoded_data);
```

Comment:
We need to think about fees here. Perhaps after this PR is merged.

We can't let smart extensions consume infinite gas. The person who initiated the transfer should be allowed to set max gas/fee.